### PR TITLE
Nano: Original PL Example code for quantization

### PR DESCRIPTION
**Pull request by zhentaocc**
Resnet 18 cifar example without quantization.  
This example is from https://github.com/PyTorchLightning/lightning-tutorials/blob/main/lightning_examples/cifar10-baseline/baseline.py
Quantization version: #3918

### CODE REVIEW

**Comment by zhentaocc** [comment]
Console output of the run:
```text
GPU available: True, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
/home/zhentaoc/anaconda3/envs/nano_torch/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py:1293: UserWarning: GPU available but not used. Set the gpus flag in your trainer `Trainer(gpus=1)` or script `--gpus=1`.
  "GPU available but not used. Set the gpus flag in your trainer"
/home/zhentaoc/anaconda3/envs/nano_torch/lib/python3.7/site-packages/pytorch_lightning/trainer/configuration_validator.py:101: UserWarning: you defined a validation_step but have no val_dataloader. Skipping val loop
  rank_zero_warn(f"you defined a {step_name} but have no {loader_name}. Skipping {stage} loop")
Epoch 0:   0%|          | 0/390 [00:00<00:00, 6990.51it/s]  
  | Name  | Type             | Params
-------------------------------------------
0 | model | ResNet           | 11.2 M
1 | loss  | CrossEntropyLoss | 0     
-------------------------------------------
11.2 M    Trainable params
0         Non-trainable params
11.2 M    Total params
44.727    Total estimated model params size (MB)
Epoch 39: 100%|██████████| 390/390 [06:56<00:00,  1.06s/it, loss=0.213, v_num=2]
Training whole time: 16605 s
Testing: 0it [00:00, ?it/s]
Testing:  99%|█████████▊| 78/79 [00:36<00:00,  2.18it/s]--------------------------------------------------------------------------------
DATALOADER:0 TEST RESULTS
{'test/0_Accuracy': 0.8306999802589417,
 'test/1_Accuracy': 0.9901999831199646,
 'test/loss': 0.5505761504173279}
--------------------------------------------------------------------------------
Testing: 100%|██████████| 79/79 [00:36<00:00,  2.16it/s]
FP32 model size: 44.804 MB
Throughput: 36 s
Accuracy: top1=83.07% top5=99.02 %

Process finished with exit code 0
```

**Comment by jason-dai** [comment]
Is this the original PL code? Can you share the original link?

**Comment by zhentaocc** [comment]
> Is this the original PL code? Can you share the original link?

I actually wrote this example from scratch. I happened to find one similar example from PL as below. Shall I use PL examples?
https://colab.research.google.com/github/PytorchLightning/lightning-tutorials/blob/publication/.notebooks/lightning_examples/cifar10-baseline.ipynb#scrollTo=cf38e878

**Comment by zhentaocc** [comment]
I have two options to choose from to convert as a quantization example:

1. https://github.com/PyTorchLightning/lightning-tutorials/blob/main/lightning_examples/cifar10-baseline/baseline.py
2. https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/basic_examples/mnist_examples/image_classifier_4_lightning_module.py

First one is from tutorial. Second one is from pl_examples, it seems much less straightforward, wrapped with a `LightningCLI` to help initialize model and trainer. From my view, second one will be a bit confusing if uses are not that familiar with PL. But I don't know if first one from tutorial is suitable to be adapted into our repo, since I don't see header in provided `baseline.py`. Any preference? @jason-dai

**Comment by jason-dai** [comment]
> I have two options to choose from to convert as a quantization example:
> 
> 1. [PyTorchLightning/lightning-tutorials@`main`/lightning_examples/cifar10-baseline/baseline.py](https://github.com/PyTorchLightning/lightning-tutorials/blob/main/lightning_examples/cifar10-baseline/baseline.py?rgh-link-date=2022-01-26T03%3A10%3A02Z)
> 2. [PyTorchLightning/pytorch-lightning@`master`/pl_examples/basic_examples/mnist_examples/image_classifier_4_lightning_module.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/basic_examples/mnist_examples/image_classifier_4_lightning_module.py?rgh-link-date=2022-01-26T03%3A10%3A02Z)
> 
> First one is from tutorial. Second one is from pl_examples, it seems much less straightforward, wrapped with a `LightningCLI` to help initialize model and trainer. From my view, second one will be a bit confusing if uses are not that familiar with PL. But I don't know if first one from tutorial is suitable to be adapted into our repo, since I don't see header in provided `baseline.py`. Any preference? @jason-dai

The first one is fine; https://github.com/PyTorchLightning/lightning-tutorials is using Apache 2.0 license

**Comment by zhentaocc** [comment]
Original PL example refers to https://github.com/PyTorchLightning/lightning-tutorials/blob/main/lightning_examples/cifar10-baseline/baseline.py?rgh-link-date=2022-01-26T03%3A10%3A02Z  

Modifications to original example:

1.  There are two parts in original example, I removed swa optimization part to focus on later changes and shorten the example.  
2.  Original script is for gpu, so I did tiny modifications to make it work on cpu.
3. Add installation message to warn users to install lightning_bolts

@jason-dai @yangw1234 Seems good? I guess in this PR I just modify examples/ so I don't need to run Jenkins. Right?

File: python/nano/example/pytorch/quantization/inc/resnet18_cifar.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+from pytorch_lightning.callbacks import LearningRateMonitor
+from pytorch_lightning.loggers import TensorBoardLogger
+from torch.optim.lr_scheduler import OneCycleLR
+from torchmetrics.functional import accuracy
+
+try:
+    from pl_bolts.datamodules import CIFAR10DataModule
+    from pl_bolts.transforms.dataset_normalizations import cifar10_normalization
+except ModuleNotFoundError:
+    raise ImportError("This example need lightning-bolts, please install by: "
```
Comment (yangw1234) [inline python/nano/example/pytorch/quantization/inc/resnet18_cifar.py:50]: I think you can just add installing lightning-bolts in this examples' readme file and not modify the code.

Comment (zhentaocc) [reply python/nano/example/pytorch/quantization/inc/resnet18_cifar.py]: Modified.

**Comment by yangw1234** [comment]
LGTM

**Pull request merged**